scripts/ci/check_compliance.py

```diff
@@ -1060,6 +1060,28 @@ def run(self):
                 self.failure(failure)
 
 
+class BinaryFiles(ComplianceTest):
+    """
+    Check that the diff contains no binary files.
+    """
+    name = "BinaryFiles"
+    doc = "No binary files allowed."
+    path_hint = "<git-top>"
+
+    def run(self):
+        BINARY_ALLOW_PATHS = ("doc/", "boards/")
```

Comment:
we can have samples with images in their docs as well

Reply:
Oh good point, added.